### PR TITLE
Improve RAPIDS support (require GPU, fix usage of GPU in parallel)

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
#1845 

*Description of changes:*
- Improve RAPIDS support (require GPU, fix usage of GPU in parallel)
- Clean up RAPIDS code by adding RapidsModelMixin

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tabular/src/autogluon/tabular/models/_utils/rapids_utils.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from typing import Dict
+
+from autogluon.core.utils import get_gpu_count_torch
+
+
+class RapidsModelMixin:
```
Comment (yinweisu) [inline tabular/src/autogluon/tabular/models/_utils/rapids_utils.py:6]: Is this model supposed to inherit `AbstractModel`? Seen couple super() calls, but this model is not inheriting any class.

Comment (Innixma) [reply tabular/src/autogluon/tabular/models/_utils/rapids_utils.py:6]: I don't believe Mixin's are supposed to inherit. @gradientsky thoughts?

Comment (gradientsky) [reply tabular/src/autogluon/tabular/models/_utils/rapids_utils.py:6]: Yes, they not supposed to be inherit `AbstractModel`, it's an additional functionality mixed-into the main class via multiple inheritance. One thing: I see some versions implement `ABC`: https://docs.python.org/3/library/abc.html#abc.ABCMeta. Not sure if that's really necessary though.

**Review by gradientsky — APPROVED** [review]

File: tabular/src/autogluon/tabular/models/_utils/rapids_utils.py (hunk 1 above)
Comment (Innixma) [reply tabular/src/autogluon/tabular/models/_utils/rapids_utils.py:6]: I don't think ABC is necessary, I tested with a dummy model using inheritance and it worked fine (tbh I'm unsure what ABC does and when it is necessary).

**Pull request merged**